### ISSUE TITLE
NVDA alpha-28430:The confirmation checkbox for NVDA/Advanced Classification cannot be selected

### ISSUE THREAD

**Issue by dpy013** [nvaccess/nvda#15005]
### Steps to reproduce:
First open NVDA menu/options/settings
Then find Advanced in the category and select I understand that changing these settings may cause NVDA to function incorrectly. check box not checked
Check the above checkbox and confirm to re-open NVDA Menu/Options/Settings/Advanced. The problem will be reproduced.
### Actual behavior:

The checkbox cannot be selected

### Expected behavior:

Check box can be selected normally

### NVDA logs, crash dumps and other attachments:
[nvda.log](https://github.com/nvaccess/nvda/files/11754569/nvda.log)

#### NVDA version:
NVDA alpha-28430

#### Windows version:

Windows 11 22H2 (10.0.22621)

#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
none
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by nvdaes** [comment]
I cannot reproduce this with a portable copy with NVDA in Spanish.

**Comment by dpy013** [comment]
> I cannot reproduce this with a portable copy with NVDA in Spanish.
The test was done with the NVDA alpha version of the mounting board

**Comment by dpy013** [comment]
Just did a test with a portable version of NVDA. No error message does appear, but the warning checkbox in settings/advanced is still unselectable

**Comment by XLTechie** [comment]
What is mounting board?

**Comment by XLTechie** [comment]
@dpy013 can you switch your log level to debug (in general settings), and reproduce this?

I can not reproduce this with the same alpha version, in English. I wonder if it is specific to the language: @cary-rowen do you also use the language of the log, and can you test this?

**Comment by XLTechie** [comment]
@dpy013 Can you set your NVDA language to English and determine if this can be reproduced then?

**Comment by cary-rowen** [comment]
I can't reproduce this with the same version of NVDA in Simplified Chinese system.

**Comment by CyrilleB79** [comment]
@dpy013, I have had a look at the log. There does not seem to be any error (please tell me if you have had an error).

From the log I can see that:
* You go to advanced settings
* You check the checkbox "I understand..."
* You tab 2 or 3 times to check that the controls in the panel can be accessed (but do not modify any one of them)
* You shift+tab until you reach "OK" button
* You open again advanced settings panel
* you tab to the checkbox "I understand..", which is unchecked. It is expected that this checkbox is unchecked when you close and reopen the settings window, do you agree? Also you do not press space to check it at this moment.
* You tab to the "Restore defaults" button, which is focusable no matter the state of the previous checkbox; this too is expected: this is the only control of the panel which is enabled no matter the state of the checkbox above.
* You shift+tab back and press Escape to exit; you do not check further the controls in the panel.

Do you agree with this description? If yes, where is the problem? From what I can see:
* The checkbox "I understand..." can be checked when you check it (the first thime you open the settings window)
* But you do not press space again when you open the settings window the second time


* You press Enter to validate

**Comment by cary-rowen** [comment]
Maybe you can get in touch with me and describe your problems in Chinese.

**Comment by cary-rowen** [comment]
Hi @XLTechie 

I communicated with @dpy013 about this issue and I think it's a misunderstanding.

@CyrilleB79 wrote:
> Maybe something that need to be clarified is that the state of checkbox "I understand..." is not saved in your config. It's a control that need to be re-checked each time you access the advanced settings panel.

You are correct, @dpy013 thought that none of the advanced options would take effect unless the checkbox was ticked.

Thanks

**Comment by XLTechie** [comment]
Thanks for your help @cary-rowen.

**Comment by CyrilleB79** [comment]
@cary-rowen, @dpy013 do you think that something needs to be clarified in the GUI or in the User Guide at NVDA core level? Or in the Chinese translation of the GUI and/or of the User Guide?

**Comment by dpy013** [comment]
The problem is solved.
I misunderstood that I was aware that changing these settings might cause NVDA to not function properly
This checkbox was misunderstood.
I always thought that the checkbox was checkable, but the switch was limited to hiding advanced features.
But this switch is only limited to hide advanced functions.
It does not record the check status.
Thank you all, I'm sorry to bother you.
